tests/unit_tests/test_raptor/test_worker.py

```diff
@@ -111,5 +116,45 @@ def test_call(self, mocked_init, mocked_Logger):
         self.assertEqual(exc, (None, None))
 
 
+    # --------------------------------------------------------------------------
+    #
+    @mock.patch.object(DefaultWorker, '__init__', return_value=None)
+    @mock.patch('radical.utils.Logger')
+    def test_sandbox(self, mocked_init, mocked_Logger):
+
+        # FIXME: this test is still invalid: `_dispatch_func()` happens after
+        #        the change to the task sandbox and as such the test will not
+        #        see sandbox change.  But testing the outer loop is (a)
+        #        different for both workers, and (b) involves spawning new
+        #        processes which I do not know how to handle / mock here in
+        #        pytest.
+
+        component = DefaultWorker()
+        component.check_pwd = os.getcwd
+        component._prof = mock.Mock()
+        component._log  = mocked_Logger
+        task = {'uid'        : 'task.0000',
+                'description': {
+                        'function': 'check_pwd',
+                        'sandbox' : '/tmp',
+                        'args'    : [],
+                        'kwargs'  : {}}}
+        out, err, ret, val, exc = component._dispatch_func(task)
```

Comment:
we should test `component._dispatch()` method here,
and for that we can patch `mp.Process` with `side_effect` as some method  that will receive `_worker_proc` and executes it.

Reply:
@mtitov , would it be imprudent to ask if you can take a stab at this?

Reply:
not at all, ok, will take car of it

Reply:
FWIW, another option would be to make `_dispatch` a wrapper around, say, dispatch_2 like this:
```py
def _dispatch(self, ...):
    return self._dispatch_2(...)
```
which would make `_dispatch_2` testable w/o the need for side effects.  Would that simplify things IYO?

Reply:
tests are updated (and `sys.exit` is considered within tests, but a corresponding question above is still valid)

Reply:
> is it expected that _dispatch always finishes with sys.exit?

That is indeed the case for the `DefaultWorker`: it spawns a new process for each task and we use `sys.exit()` to reliably terminate that process again.  The `MPIWorker` though does not spawn new processes and thus does not  use `sys.exit`.

Reply:
Thanks for the test update!!

Reply:
> > is it expected that _dispatch always finishes with sys.exit?
> 
> That is indeed the case for the `DefaultWorker`: it spawns a new process for each task and we use `sys.exit()` to reliably terminate that process again. The `MPIWorker` though does not spawn new processes and thus does not use `sys.exit`.

ah, I see, that makes sense, thank you!